docs/contributing/components.md

```diff
@@ -0,0 +1,114 @@
+---
+sidebar_label: Components
+---
+
+```mdx-code-block
+import CodeBlock from '@theme/CodeBlock';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+```
+
+# Contributing to Components
+
+The source code for Resoto lives in the [`someengineering/resoto` repository on GitHub](https://github.com/someengineering/resoto).
+Resoto consists of multiple components, where each component is maintained as separate project.
+A change should target one of the components.
+
+- [Resoto Core](https://github.com/someengineering/resoto/tree/main/resotocore)
+- [Resoto Shell](https://github.com/someengineering/resoto/tree/main/resotoshell)
+- [Resoto Worker](https://github.com/someengineering/resoto/tree/main/resotoworker)
+- [Resoto Metrics](https://github.com/someengineering/resoto/tree/main/resotometrics)
+- [Resoto Library](https://github.com/someengineering/resoto/tree/main/resotolib)
+
+## Authoring Changes
+
+Contributions are made via [pull requests to the GitHub repository](https://github.com/someengineering/resoto/pulls).
+You will first need to [fork](https://docs.github.com/get-started/quickstart/fork-a-repo) the repository.
+
+#### Prerequisites
+
+- [Git](https://git-scm.com)
+- [Python](https://python.org) 3.8+ (3.10 is recommended)
+- [ArangoDB](https://arangodb.com) 3.8.2+
+- [GNU Compiler Collection (GCC)](https://gcc.gnu.org) (depending on the host system, Python dependencies may need to be compiled from source)
+
+#### Setting Up a Virtual Environment
+
+We recommend using a [Python virtual environment](https://docs.python.org/3/tutorial/venv.html).
+
+A script is provided to simplify the process of configuring the virtual environment:
+
+```bash
+./setup_venv.sh --dev --path .
+```
+
+Activate the virtual environment:
+
+```mdx-code-block
+<Tabs>
+  <TabItem value="linux" label="Linux/macOS">
+    <CodeBlock language="bash">source venv/bin/activate</CodeBlock>
+  </TabItem>
+  <TabItem value="windows" label="Windows">
+    <CodeBlock language="powershell">venv\Scripts\activate.bat</CodeBlock>
+  </TabItem>
+</Tabs>
+```
+
+#### Starting the Database
+
+After you have installed ArangoDB, you need to start it.
+The way the database is started depends on the way it has been installed (`systemctl` on Linux, clicking the application icon on macOS, etc.).
+
+**The expected default access is the built-in user account `root` with a configured empty string as password.**
+No additional settings are required in such a setup, since `resotocore` will use this access to create all necessary items.
+This setting is required for tests to pass, as the tests create users and databases.
```

Comment:
```suggestion
Start ArangoDB (using `systemctl` on Linux, by clicking the application icon in macOS, etc.).

**By default, the built-in user account `root` does not have a password configured.** This is necessary for `resotocore` to perform the database setup, and also required for tests to pass.
```
Good to merge after this!

Reply:
ah sorry - the change was not committed. one sec